client/src/app.js

```diff
@@ -191,15 +208,14 @@ export default function main({ DOM, HTTP, route, storage, scanner: scan$, search
 
   // Transaction analysis
   , txAnalysis$ = tx$.filter(Boolean)
-      .combineLatest(mempool$, feeEst$, (tx, mempool, feeEst) =>
-          ({ tx, mempool, feeEst, feerate: tx.fee ? tx.fee / tx.weight * 4 : null }))
-      .map(({ tx, feerate, mempool, feeEst }) => ({
+      .combineLatest(mempool$, feeEst$, calculateFeerates)
+      .map(({ tx, feerate, mempool, feeEst, discountFeerate}) => ({
         feerate
       , privacyAnalysis: getPrivacyAnalysis(tx)
       , segwitGains: calcSegwitFeeGains(tx)
       , mempoolDepth: !tx.status.confirmed && feerate != null && mempool ? getMempoolDepth(mempool.fee_histogram, feerate) : null
       , confEstimate: !tx.status.confirmed && feerate != null && feeEst ? getConfEstimate(feeEst, feerate) : null
```

Comment:
Shouldn't `mempoolDepth` and `confEstimate` use the `discountedFeerate` too?

It might be more convenient to implement this if `calculateFeerates` added an `effectiveFeerate` field that is either the `discountFeerate` when available or the regular `feerate` otherwise.

Then we can use it for `calculateOverpayment()`, `getMempoolDepth()` and `getConfEstimate()` instead of having each function deal with both (as `calculateOverpayment()` currently does).

Reply:
I have followed this uggestion and added it into an additional commit. I will squash after review feedback.